packages/manager/apps/catalog-revamp/src/utils/utils.ts

```diff
@@ -0,0 +1,119 @@
+export type Product = {
+  id: number;
+  name: string;
+  description: string;
+  lang: string;
+  categories: string[];
+  url: string;
+  regionTags: string[];
+  productName: string;
+  order: string;
+  universe: string;
+  category: string;
+};
+
+type ProductKey = 'universe' | 'category' | 'name';
+
+export const groupItemsByUniverse = (itemsToGroup: Product[]) => {
+  return itemsToGroup.reduce<Record<string, Product[]>>((acc, item) => {
+    if (!acc[item.universe]) acc[item.universe] = [];
+    acc[item.universe].push(item);
+    return acc;
+  }, {});
+};
+
+export const filterByProperty = (
+  product: Product,
+  query: string[],
+  property: ProductKey,
+) => {
+  return query.length === 0 || query.includes(product[property]);
+};
+
+export const getAvailableCategories = (
+  products: Product[],
+  selectedUniverses: string[],
+) => {
+  const itemsFiltered = products.filter((product) =>
+    filterByProperty(product, selectedUniverses, 'universe'),
+  );
+  const uniqueCategories = [
+    ...new Set(itemsFiltered.map((item) => item.category)),
+  ];
+  return uniqueCategories.sort((a, b) =>
+    a.localeCompare(b, undefined, { sensitivity: 'base' }),
+  );
+};
+
+export const matchSearchText = (product: Product, searchText: string) => {
+  const search = searchText?.toLowerCase();
+  return (
+    !search ||
+    product.category.toLowerCase().includes(search) ||
+    product.universe.toLowerCase().includes(search) ||
+    product.name.toLowerCase().includes(search)
+  );
+};
+
+export const filterProducts = (
+  products: Product[],
+  selectedCategories: string[],
+  selectedUniverses: string[],
+  searchText: string,
+) => {
+  return products.filter(
+    (product) =>
+      filterByProperty(product, selectedCategories, 'category') &&
+      filterByProperty(product, selectedUniverses, 'universe') &&
+      matchSearchText(product, searchText),
+  );
+};
+
+const countAndFormat = (objectList: any[], property: string) => {
+  const countByProperty = objectList.reduce((acc, obj) => {
+    acc[obj[property]] = (acc[obj[property]] || 0) + 1;
+    return acc;
+  }, {} as Record<string, number>);
+
+  return Object.entries(countByProperty)
+    .map(([key, count]) => ({ [property]: key, count }))
+    .sort((a, b) => {
+      // Type assertion to clarify that a[property] and b[property] are strings
+      const aValue = a[property] as string;
+      const bValue = b[property] as string;
+      return aValue.localeCompare(bValue, undefined, { sensitivity: 'base' });
+    });
```

Comment:
```suggestion
    .sort((a, b) => (a[property] as string).localeCompare(b[property] as string, undefined, { sensitivity: 'base' });
```

We don't need a comment, reading the code is self-explanatory

Reply:
Done, thanks